File: AutoIndustryMarketPlace/lib/models/part.dart
class Part {
  final String id;
  final String name;
  final String category;
  final double price;
  final int stock;

  Part({
    required this.id,
    required this.name,
    required this.category,
    required this.price,
    required this.stock,
  });

  factory Part.fromJson(Map<String, dynamic> json) {
    return Part(
      id: json['id'],
      name: json['name'],
      category: json['category'],
      price: json['price'],
      stock: json['stock'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'category': category,
      'price': price,
      'stock': stock,
    };
  }
}

File: Autoindustry-Backend/src/main/java/com/autoindustry/backend/model/Part.java
package com.autoindustry.backend.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "parts")
@Data
public class Part {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String category;

    private double price;

    private int stock;
}